server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -417,7 +459,7 @@ public void doRun()
               synchronized (hydrant) {
                 if (!hydrant.hasSwapped()) {
                   log.info("Hydrant[%s] hasn't swapped yet, swapping. Sink[%s]", hydrant, sink);
-                  final int rowCount = persistHydrant(hydrant, schema, interval);
+                  final int rowCount = persistHydrant(hydrant, schema, interval, null);
```

Comment:
I agree but there's a lot of leeway in what "fully functional" means :)

This PR was never meant to support scale up/down or restoration of offsets after loss of local disk (as opposed to temporary outage). So it was never meant to be perfect or even a strict improvement over the old kafka ingestion. I think that good questions are, does it provide useful behavior and is it a step in the right direction?

My take is that the behavior is not the most useful in the world (it doesn't fix the data dupe/loss or HA issues of the existing kafka ingestion, and loses the ability to easily scale up and down). But the new Firehose is better, and storing metadata along with segments is good, so I think it's a step in the right direction.